core/blockchain.go

```diff
@@ -1773,6 +1776,10 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals, setHead bool)
 
 		// Retrieve the parent block and it's state to execute on top
 		start := time.Now()
+		//begin PluGeth code injection
+		blockExecutionLatencyTimer.Update(time.Since(time.Unix(int64(block.Header().Time), 0))) // Duration since block timestamp before block execution
```

Comment:
This metrics capture the "latency" between the block creation and the execution in plugeth. If you improve the p2p you should see a change here